### PR TITLE
feat(Storybook): Initial project config and main qml/cpp + sample page

### PR DESCRIPTION
**Pull request by micieslak**
### What does the PR do

- Initial project config and main qml/cpp
- First sample page (CommunitiesPortalLayout)

The configuration allows easily substitute real store implementations with empty stubs. Then stub types can be used to mock stores. Mocked store can interact with helper components to allow reaching various states and edge cases of the component.

Closes: #7547 

### Affected areas

Storybook

### Screenshot of functionality (including design for comparison)

[Kazam_screencast_00034.webm](https://user-images.githubusercontent.com/20650004/194008259-386d294b-7dcd-48fe-8469-1ce1d7879ebe.webm)

### CODE REVIEW

**Comment by alexandraB99** [comment]
I think storybook needs a README

**Comment by micieslak** [comment]
> I think storybook needs a README

Could we postpone a bit creating README for that (at least not in this PR)? There is still high chance of changes, including assumptions how it should work.

File: storybook/main.qml [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+        leftPanel: StatusScrollView {
+            anchors.fill: parent
+            anchors.topMargin: 48
+
+            Column {
+                id: navigation
+                spacing: 0
+
+                StatusNavigationListItem {
+                    title: "CommunitiesPortal"
```
Comment (caybro) [inline storybook/main.qml:41]: "CommunitiesPortalLayout" pretty pls :) This should include the name of the real component

File: storybook/main.cpp [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+int main(int argc, char *argv[])
+{
+#if QT_VERSION < QT_VERSION_CHECK(6, 0, 0)
+    QCoreApplication::setAttribute(Qt::AA_EnableHighDpiScaling);
+#endif
+    QGuiApplication app(argc, argv);
+    app.setOrganizationName("Status");
+    app.setOrganizationDomain("status.im");
+    app.setApplicationName("Status Dekstop Storybook");
```
Comment (caybro) [inline storybook/main.cpp:13]: Typo here

File: storybook/main.qml [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+                    selected: viewLoader.source.toString().includes(title)
```
Comment (caybro) [inline storybook/main.qml:42]: ...which won't of course work here but you can ~~ab~~use the `objectName` property for that

File: storybook/main.qml [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+                    onClicked: mainPageView.page(title);
```
Comment (caybro) [inline storybook/main.qml:43]: Same here

File: storybook/pages/CommunitiesPortalPage.qml [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+
+                filters: IndexFilter {
+                    inverted: true
+                    minimumIndex: Math.floor(slider.value)
+                }
+            }
+        }
+    }
+
+    Rectangle {
```
Comment (caybro) [inline storybook/pages/CommunitiesPortalPage.qml:35]: StatusDialogDivider

File: storybook/main.qml [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+
+
+ApplicationWindow {
+    id: rootWindow
+
+    width: 1450
+    height: 840
+    visible: true
+
+    title: qsTr("Status Desktop Storybook App")
```
Comment (caybro) [inline storybook/main.qml:18]: You don't need this; it's either taken from "applicationName" or "applicationDisplayName" (see main.cpp)

File: storybook/main.qml [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+
+            Loader {
+                id: viewLoader
+                anchors.fill: parent
+                clip: true
+                source: storeSettings.selected
+            }
+        }
+    }
+
```
Comment (caybro) [inline storybook/main.qml:60]: I'd like to have some sort of "Light/Dark" mode switch somewhere (not necessarily in this PR)

**Review by caybro — CHANGES_REQUESTED** [review]
Looks good and really useful, just some minor nitpicks

File: storybook/main.qml (hunk 7 above)
Comment (micieslak) [reply storybook/main.qml:58]: Yeah, me too, but in next PRs :)

**Review by caybro — APPROVED** [review]

**Review by noeliaSD — APPROVED** [review]
Great job and good starting point!

Just a minor comment.. From my point of view, I would start with components like `StatusCommunityCard` instead of a complete desktop page like `CommunityPortalLayout`. 

Excited to see what comes next! ;)

**Comment by micieslak** [comment]
> Great job and good starting point!
> 
> Just a minor comment.. From my point of view, I would start with components like `StatusCommunityCard` instead of a complete desktop page like `CommunityPortalLayout`.
> 
> Excited to see what comes next! ;)

Thank you @noeliaSD for the review and feedback! I think that a separate page for `StatusCommunityCard` may not be needed, because it will be possible to go through all states/properties of `StatusCommunityCard` through page for `CommunityPortalLayout`. Let me create one more PR and we will go back to the discussion about that :)

**Comment by noeliaSD** [comment]
> 

Sure! I'm not saying that a new page specifically for cards is needed, I'm saying that I had started with a simple card instead of replicating the entire communities portal design.

**Pull request merged**